packages/@n8n/api-types/src/dto/user/users-list-filter.dto.ts

```diff
@@ -1,25 +1,137 @@
+import { jsonParse } from 'n8n-workflow';
 import { z } from 'zod';
 import { Z } from 'zod-class';
 
 import { paginationSchema } from '../pagination/pagination.dto';
 
 const USERS_LIST_SORT_OPTIONS = [
-	'name:asc',
-	'name:desc',
+	'firstName:asc',
+	'firstName:desc',
+	'lastName:asc',
+	'lastName:desc',
 	'role:asc', // ascending order by role is Owner, Admin, Member
 	'role:desc',
-	'lastActive:asc',
-	'lastActive:desc',
+	// 'lastActive:asc',
+	// 'lastActive:desc',
 ] as const;
 
+const createTakeValidator = (maxItems: number) =>
```

Comment:
There already is a createTakeValidator functon (exported) in the pagination.dto.ts. Any reason to redefine it here ?

Reply:
Yes, because we want to support negative numbers here to get all users. The existing one doesn't support that.

Reply:
Ah ! I see :thinking: 
It might be worth checking if we could make this rule universal (thus changing the original createTakeValidator.

Reply:
I'm not sure we want to do this everywhere.

Reply:
Getting all workflow executions, could be very bad and why would we have a maximum for take in that case.